src/libraries/System.Threading.Tasks.Dataflow/src/Blocks/TransformManyBlock.cs

```diff
@@ -86,24 +89,24 @@ public sealed class TransformManyBlock<TInput, TOutput> : IPropagatorBlock<TInpu
         /// <param name="dataflowBlockOptions">The options with which to configure this <see cref="TransformManyBlock{TInput,TOutput}"/>.</param>
         /// <exception cref="System.ArgumentNullException">The <paramref name="transform"/> is null (Nothing in Visual Basic).</exception>
         /// <exception cref="System.ArgumentNullException">The <paramref name="dataflowBlockOptions"/> is null (Nothing in Visual Basic).</exception>
-        public TransformManyBlock(Func<TInput, Task<IEnumerable<TOutput>>> transform, ExecutionDataflowBlockOptions dataflowBlockOptions) :
-            this(null, transform, dataflowBlockOptions)
-        { }
+        public TransformManyBlock(Func<TInput, Task<IEnumerable<TOutput>>> transform, ExecutionDataflowBlockOptions dataflowBlockOptions)
+        {
+            // Validate arguments.
+            if (transform == null) throw new ArgumentNullException(nameof(transform));
+            Initialize(messageId => ProcessMessageWithTask(transform, messageId), dataflowBlockOptions, ref _source!, ref _target!, ref _reorderingBuffer, async: true);
```

Comment:
what are the benefits of changing private `ctor` to a method? You need to pass few more extra args, take references and use `!` to pretend that fields which are nulls are not nulls. I don't like this part of the refactor

Reply:
It is an attempt to solve an issue reported by @stephentoub during a previous review in https://github.com/dotnet/runtime/pull/37876#discussion_r439893214
To duplicate code or not to duplicate code... that is the question!